mmf/utils/checkpoint.py

```diff
@@ -379,11 +384,15 @@ def _get_vcs_fields(self):
             "git/diff": self.git_repo.git.diff("--no-prefix"),
         }
 
+    def save_func(self, *args):
+        return xm.save(*args) if is_xla() else torch.save(*args)
+
     def save(self, update, iteration=None, update_best=False):
         # Only save in main process
-        if not is_master():
+        if not is_master() and not is_xla():
```

Comment:
Does this mean all processes in xla will save checkpoints? I am not familiar with xla internal semantics and may be missing something here, but is this the expected behavior?

Reply:
Added comment explaining the rationale. for xla we use xm.save which saves only the checkpoint only for master and does the synchronization as well. if we run it only only for master, it will result in synchronization lock because other cores wont run the same code.